CMakeLists.txt

```diff
@@ -545,7 +545,7 @@ message(STATUS "Using Lua interpreter: ${LUA_PRG}")
 
 option(COMPILE_LUA "Pre-compile Lua sources into bytecode (for sources that are included in the binary)" ON)
 
-if(COMPILE_LUA AND NOT WIN32)
+if(COMPILE_LUA AND NOT CMAKE_BUILD_TYPE STREQUAL "Debug" AND NOT WIN32)
```

Comment:
This makes it impossible for someone to compile byte code on a debug build even if they wanted to. Instead, we should set the default value of COMPILE_LUA based on build type.

Debug: COMPILE_LUA = OFF
else: COMPILE_LUA = ON

This way it’s still user-overridable.